File: src/ast/GlobalQualifiedName.java
package ast;

import java.io.StringWriter;

public class GlobalQualifiedName extends ParentNode {
    public GlobalQualifiedName(BaseNode child) {
        super(NodeType.GlobalQualifiedName, child);
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("::");
        child.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/GlobalQualifiedName.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class GlobalQualifiedName(child: BaseNode?) : ParentNode(NodeType.GlobalQualifiedName, child) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("::")
        child!!.print(writer)
    }
}